context/context.go

```diff
@@ -123,15 +130,22 @@ func (context *AMFContext) TmsiAllocate() int32 {
 }
 
 func (context *AMFContext) AllocateAmfUeNgapID() (int64, error) {
-	// val, err := AllocateUniqueID(&amfUeNGAPIDGenerator, "amfUeNgapID")
-	val, err := context.Drsm.AllocateInt32ID()
+	var val int64
+	var err error
+	if context.EnableDbStore {
+		var tmp int32
+		tmp, err = context.Drsm.AllocateInt32ID()
+		val = int64(tmp)
+	} else {
+		val, err = AllocateUniqueID(&amfUeNGAPIDGenerator, "amfUeNgapID")
+	}
 	if err != nil {
 		logger.ContextLog.Errorf("Allocate NgapID error: %+v", err)
 		return -1, err
 	}
 
 	logger.ContextLog.Infof("Allocate AmfUeNgapID : %v", val)
-	return int64(val), nil
+	return val, nil
```

Comment:
@gruyaume I am not sure if adding these functions make sense, as they require a whole new `IDGeneratorInt32`, as all the constructor and functions expect `int64` as parameters and return values (see [idgenerator](https://github.com/omec-project/util/blob/main/idgenerator/idgenerator.go) library for reference).
Also, in lines 46-47 the generators are instantiated with a range between 1 and `MaxInt32` which implies safe casting from `int64` to `int32`. What do you think?

Reply:
I suggested above an approach with the `ok` semantics that could at least catch errors if they do happen.

Reply:
Probably a better approach in the future would be to not use base types here and create a specific type to handles IDs. This would probably be a more involved refactoring however, as this would mean that everywhere an ID is used, we would have to change signatures.